AddRoutineView.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  AddRoutineView.swift
+//  OnTrack
+//
+//  Allows user to add a new routine to their list of routines.
+//
+//  Created by Noah Sadir on 4/4/24.
+//
+//  Contributors: David
+//
+
+import SwiftUI
+
+struct AddRoutineView: View {
+    @State var id: String = ""
+    @State var title: String = ""
+    @State var description: String = ""
+    @State var category: String = ""
+    @State var frequency: Int = 0
+    @State var dates: String = ""
+    @State var streakLength: Int = 0
+    
+    @StateObject var vm = ViewModel()
```

Comment:
View model needs to be passed as an environment object.

This was already included in the code provided.